src/plugins/controls/public/react_controls/control_group/get_control_group_factory.tsx

```diff
@@ -133,7 +134,7 @@ export const getControlGroupEmbeddableFactory = (services: {
               controlsManager.api.children$
             ),
             controlGroupFetch$(ignoreParentSettings$, parentApi ? parentApi : {})
-          ),
+          ).pipe(distinctUntilChanged((a, b) => deepEqual(a, b))),
```

Comment:
Previously, `controlFetch$` would emit the same value multiple times as things loaded - now, we should only emit when things change. This is what **actually** stops the controls from fetching multiple times unnecessarily as things load - the change in `chaining$` was just a small improvement that I noticed.